App/Sagas/TextileSagas.js

```diff
@@ -146,6 +151,13 @@ export function * photosTask (action) {
 }
 
 export function * removePayloadFile ({data}) {
-  const {file} = data
-  yield call(RNFS.unlink, file)
+  const { uploadId } = data
+  const items = yield select(TextileSelectors.items)
```

Comment:
Yea seems like this would be cleaner if there were no `cleanup` state, and this saga was fired after an image was successfully uploaded (like it was configured before). The Selector in here could just return the file path for the given upload id. Then the complexity of filtering state looking for the image would be in the selector and this could remain simple like: 
```
const { uploadId } = data
const file = yield select(TextileSelectors.fileForUploadId, uploadId)
yield call(RNFS.unlink, file)
```

Reply:
it will fire before the original handleImageUploadComplete, so then there wasn't a record of the path in the items that it could get